File: src/main/java/com/kairlec/service/DescriptionMapService.java
package com.kairlec.service;

import com.kairlec.pojo.DescriptionMap;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface DescriptionMapService {

    Integer init();

    List<DescriptionMap> getAll();

    DescriptionMap getDescriptionMap(String path);

    List<DescriptionMap> getSection(List<String> Paths);

    Integer insertDescriptionMap(DescriptionMap descriptionMap);

    Integer deleteDescriptionMap(String path);

    Integer updateDescriptionMap(DescriptionMap descriptionMap);
}

File: src/main/java/com/kairlec/service/DescriptionMapService.kt
package com.kairlec.service

import com.kairlec.pojo.DescriptionMap
import org.springframework.stereotype.Service

@Service
interface DescriptionMapService {
    fun init(): Int?
    fun all(): List<DescriptionMap>?
    fun getDescriptionMap(path: String): DescriptionMap?
    fun getSection(Paths: List<String>): List<DescriptionMap>?
    fun insertDescriptionMap(descriptionMap: DescriptionMap): Int?
    fun deleteDescriptionMap(path: String): Int?
    fun updateDescriptionMap(descriptionMap: DescriptionMap): Int?
}
